securedrop/source_app/__init__.py

```diff
@@ -128,6 +108,14 @@ def setup_g() -> Optional[werkzeug.Response]:
 
         return None
 
+    @app.before_request
+    @ignore_static
+    def check_tor2web() -> Optional[Tuple[str, int]]:
+        if 'X-tor2web' in request.headers:
+            flash(gettext("Tor2Web proxies do not protect your anonymity!"), "error")
```

Comment:
Can we refactor out these two lines, the flash and template into a separate function? Then I can reuse it in my PR rather than redirecting to `/tor2web-warning`.

Reply:
went a different route (:drum:) - /tor2web-warning now has 403 status and this check redirects to it.